feat(*) exit if parent died

gszr opened the pull request:
Run a goroutine every minute to check if parent died and exit.
Assumption: if parent dies, the process gets adopted by init (pid 1).

javierguerragiraldez commented:
i think this code is nice, but i fail to see the need.  in my tests, the pluginserver always exits when the nginx worker dies, even when i messed it up and ended in a weird loop.  is your experience different? in what kind of scenario can this help?

dndx commented on main.go:
```diff
@@ -111,6 +116,16 @@ func main() {
+					os.Exit(0)
+				}
+
+				time.Sleep(1 * time.Minute)
```
I think we can poll much more frequently, what about 1 seconds? There is basically negligible performance impact from this logic. Exiting the plugin server too slowly could cause excessive memory usage under certain cases (for example, when user restarts Kong many times over a short time period).

dndx commented on main.go:
```diff
@@ -111,6 +116,16 @@ func main() {
+		go func() {
+			for {
+				if ! isParentAlive() {
+					os.Exit(0)
```
It will be better to add some logging when this happens, so we know why the server quits.

dndx commented:
@javierguerragiraldez I think worst come to worst if the plugin server didn't exit by design, then this logic provides some safety net against the worst case scenario (a zombie process). Since we do not have the capabilities to test on enough O/Ses to make sure that's always the case.

gszr commented:
> @javierguerragiraldez I think worst come to worst if the plugin server didn't exit by design, then this logic provides some safety net against the worst case scenario (a zombie process). Since we do not have the capabilities to test on enough O/Ses to make sure that's always the case.

Agreed on that.

I just changed the period to 1 second, and added a log message when the process exits.
